src/test/java/com/google/finapp/FinAppIT.java

```diff
@@ -19,68 +19,104 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.Database;
 import com.google.cloud.spanner.DatabaseClient;
-import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.IntegrationTest;
+import com.google.cloud.spanner.IntegrationTestEnv;
 import com.google.cloud.spanner.Key;
 import com.google.cloud.spanner.KeySet;
 import com.google.cloud.spanner.ResultSet;
-import com.google.cloud.spanner.Spanner;
-import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.testing.RemoteSpannerHelper;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
 import java.math.BigDecimal;
 import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
 import java.util.UUID;
-import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.ValueSource;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
 
+@Category(IntegrationTest.class)
 public class FinAppIT {
 
   private static SpannerDaoInterface JDBCDao;
   private static SpannerDaoInterface JavaDao;
   private static DatabaseClient databaseClient;
+  private static Database db;
```

Comment:
static is good for True constants (When a final field is initialized to a hard-coded constant value) and ok-ish effective constants (technically mutable, the class is careful to never modify it once it is initialized). I would just drop static from all the variables here.

Reply:
The `static` keyword seems to be required, since the later `@BeforeClass` method initializes the variables in a static context. (I get the error "non-static field cannot be referenced from a static context" if I try to remove the `static`)

Reply:
I'm ok with this solution, but for your consideration:
There's both `@BeforeClass` that is static and `@Before` that is non-static and runs after each test.
One way to think about it is:
- Objects that are expensive to construct and could be shared between tests can be done once on `@BeforeClass`. You get the static ugliness, but it's efficient. Creating a database is a good example of a potentially expensive operation
- Objects that are cheaper to construct can go to `@Before` and be non-static

Reply:
I personally would recommend using @Before here. This would help get rid of static here and solve the parameterization issues mentioned elsewhere. The objects themselves shouldn't be very expensive to construct IMO - we can try comparing the run times of the two approaches perhaps. As long as the run time isn't grossly bad, IMO the readability benefits outweigh the performance aspects.

Reply:
My only concern would db DB creation when running against real spanner. DB Creation is actually non trivial for spanner backend and there are limits in the admin api rate of calls / etc...

Indeed in the JDBC tests, db creation is the only thing that happens on `@BeforeClass` https://github.com/googleapis/java-spanner-jdbc/blob/6d545b617cd2ab21de88a77ce8e9ebe31cf088bf/src/test/java/com/google/cloud/spanner/jdbc/ITAbstractJdbcTest.java#L93

Reply:
Even `Before` is giving me the "non-static field cannot be referenced from a static context" error when I use it to try to initialize a non-static field.

Reply:
Ignore what I said previously about the "non-static..." error. 

I think it's fair to initialize the SpannerDaoInterface in `@Before`, but I'm with @voulg  in that setting up the database should be done in `@BeforeClass`

Reply:
I can remove the statics of fields initialized in `@Before`, but not `@BeforeClass`, since it seems that it is standard practice to put `static` in `@BeforeClass` method signatures. 

I'll put the database initialization in `@BeforeClass` and spanner Dao initialization in `@Before` (much less expensive, since you just pass in a client or strings)

Reply:
Actually, there is an awkwardness in the fact that you need `testHelper` to grab the project and instance IDs, and `testHelper` isn't a field of the class. So to get around this, you would either need to make `testHelper` a static variable or the project and instance IDs. It would probably be less convoluted (with very slight performance gains) to just initialize the spanner dao in `BeforeClass`

Reply:
Acknowledged, SG. Thanks!